rcl/test/rcl/test_get_actual_qos.cpp

```diff
@@ -182,34 +192,42 @@ get_parameters()
 
 #ifdef RMW_IMPLEMENTATION_STR
   std::string rmw_implementation_str = RMW_IMPLEMENTATION_STR;
-  if (!rmw_implementation_str.compare("rmw_fastrtps_cpp") ||
-    !rmw_implementation_str.compare("rmw_fastrtps_dynamic_cpp"))
+  if (rmw_implementation_str == "rmw_fastrtps_cpp" ||
+    rmw_implementation_str == "rmw_fastrtps_dynamic_cpp")
   {
     rmw_qos_profile_t expected_system_default_qos = {
       RMW_QOS_POLICY_HISTORY_KEEP_LAST,
       1,
       RMW_QOS_POLICY_RELIABILITY_RELIABLE,
       RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL,
+      RMW_QOS_DEADLINE_DEFAULT,
+      RMW_QOS_LIFESPAN_DEFAULT,
+      RMW_QOS_POLICY_LIVELINESS_AUTOMATIC,
+      RMW_QOS_LIFESPAN_DEFAULT,
       false};
     parameters.push_back({
       rmw_qos_profile_system_default,
       expected_system_default_qos,
       "publisher_system_default_qos"});
   } else {
-    if (!rmw_implementation_str.compare("rmw_opensplice_cpp") ||
-      !rmw_implementation_str.compare("rmw_connext_cpp") ||
-      !rmw_implementation_str.compare("rmw_connext_dynamic_cpp"))
+    if (rmw_implementation_str == "rmw_connext_cpp" ||
+        rmw_implementation_str == "rmw_connext_dynamic_cpp" ||
+        rmw_implementation_str == "rmw_opensplice_cpp")
```

Comment:
Please either undo this change, or change it to be like this:

```c++
    if (
      rmw_implementation_str == "rmw_connext_cpp" ||
      rmw_implementation_str == "rmw_connext_dynamic_cpp" ||
      rmw_implementation_str == "rmw_opensplice_cpp")
```

We avoid indenting more than one level between any two lines, or indenting non-multiples of the indentation (e.g. 3 or 5 spaces), including doing so in order to vertically align the indention like this.

Reply:
Sure, I will change the formatting.

Reply:
Removed the extra indentation, but no longer vertically aligned. Is that okay?

Reply:
Yes, as-is is fine. I normally would place the first part of condition on its own line too, so they are vertically aligned without extra indention, e.g.:

```c++
    if (
      rmw_implementation_str == "rmw_connext_cpp" ||
      rmw_implementation_str == "rmw_connext_dynamic_cpp" ||
      rmw_implementation_str == "rmw_opensplice_cpp")
    {
      // ...
    }
```

Or if you prefer:

```c++
    if (
      rmw_implementation_str == "rmw_connext_cpp" ||
      rmw_implementation_str == "rmw_connext_dynamic_cpp" ||
      rmw_implementation_str == "rmw_opensplice_cpp"
    ) {
      // ...
    }
```

The most important thing is that if you're scanning the code you can see where different scopes start/end, e.g.:

```
// bad, is first statement an indention typo or a new scope (at first glance)?
if statement starts -------> | if (condition1 &&
                             |     condition2 &&
hard to see end here ------> |     condition3) {
hard to see body start ----> |   statement1;
                             |   statement2;
                             |   statement3;
                             | }

// good, start and end for scopes are in the same column
if statement starts -------> | if (
                             |   condition1 &&
                             |   condition2 &&
                             |   condition3)
easy to see body start ----> | {
                             |   statement1;
                             |   statement2;
                             |   statement3;
                             | }
```

I think about it like as if it were the "same as":

```c++
bool condition = (
  rmw_implementation_str == "rmw_connext_cpp" ||
  rmw_implementation_str == "rmw_connext_dynamic_cpp" ||
  rmw_implementation_str == "rmw_opensplice_cpp"
);

if (condition) {
  // ...
}
```

Which matches our style with initializer lists in c++ and dictionaries in Python, e.g.:

```
... = {
  "item1",
  "item2,
  // ...
}
```